### ISSUE TITLE
Emoji parser interferes with link parser

### ISSUE THREAD

**Issue by clacke** [friendica/friendica#5770]
### Expected behavior

A string detected as a URL becomes a link to that literal URL.

### Actual behavior

Characters in the URL may be replaced with other characters, as if the URL were plain text.

### Steps to reproduce the problem

1. Post a link to https://www.wikidata.org/wiki/Wikidata:Data_access

Expected:

Link points to https://www.wikidata.org/wiki/Wikidata:Data_access , link text is "https://www.wikidata.org/wiki/Wikidata:Data_access".

Actual:

Link points to https://www.wikidata.org/wiki/Wikidata%F0%9F%98%81ata_access , link text is "https://www.wikidata.org/wiki/Wikidata😁ata_access"

### Friendica version you encountered the problem

`This is Friendica, version 2018.08-rc that is running at the web location https://libranet.de. The database version is 1283, the post update version is 1281.`

**Comment by tobiasd** [comment]
> 1. Post a link to https://www.wikidata.org/wiki/Wikidata:Data_access
> 
> Expected:
> 
> Link points to https://www.wikidata.org/wiki/Wikidata:Data_access ,
> link text is "https://www.wikidata.org/wiki/Wikidata:Data_access".

I tested it in a private note, the link works as expected. As the actual output you observed contains a emoji, I would guess that the unicode_smilies addon that libranet is using (and I'm not) is too greedy with the replacement of 2 (`:D`) letter symbols.

**Comment by clacke** [comment]
Ok, so that's an issue against `unicode_smilies` then.

**Comment by clacke** [comment]
Seems friendica-addons/unicode_smilies has no issue tracker. :-(

**Comment by tobiasd** [comment]
> Seems friendica/friendica-addons has no issue tracker. :-(

No, you used the correct issue tracker ;-) I was just guessing the cause of the problem for who-ever will take a look at the problem later.

**Comment by guzzisti** [comment]
As i'm facing the same issue, i had a brief look through the unicode-smilies addon, which itself is quite simple. So obviously the fancy replacement stuff takes place somewhere else.

There are different ways to fix this issue that come to my mind:

1. simply remove the replacement of `:)` and `:(` from unicode_smilies, as `:-)` and  `:-(` are still there
2. require a space before the smilies, this would most likely fix the issues with uris
3. leave unicode_smilies as it is and fix the friendica module which does the replacement stuff (have to dig further through the friendica code to find the culprit :smile:  )

I'm not sure which attempt is prefered by the devs, feedback welcome.

**Comment by MrPetovan** [comment]
Our plan is to eventually use a Lexer. This would enable us to separate in a post what is a code block, what is a URL, and what is text paragraphs. We only want to replace emojis on the latter category. For now it's free for all (except code blocks) which means these cases will keep happening until we actually go forward with the lexer.

**Comment by guzzisti** [comment]
A lexer would be the premium solution but also probably the most complex. As a quick fix i simply disable all smilies "without a nose" in the addon, this should for now with the most common cases.

I'll provide a PR if you are interested.

**Comment by MrPetovan** [comment]
I'm not using this addon so I'm not sure what the expected usage is, but I have the feeling that excluding all smilies without a nose would leave a lot of smilies unreplaced. Is it worth it to avoid unwarranted replacements in URLs?
